### PR TITLE
Adds example from PerfDetective to the test suite

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

**Comment by vulder** [comment]
~@Kaufi-Jonas Can you please take a look at this PR?~
~Somehow, I cannot add you as a reviewer :roll_eyes:~

File: src/PerfDetectiveExample/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+add_executable(
+  PerfDetectiveExample
+
+  PDEmain.cpp
+)
+
+set_target_properties(MultiSharedMultipleRegions PROPERTIES
+  CXX_STANDARD 17
+  CXX_STANDARD_REQUIRED ON
+)
```
Comment (vulder) [inline src/PerfDetectiveExample/CMakeLists.txt]:
```suggestion
```

File: src/PerfDetectiveExample/PDEmain.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#include "fpcsc/perf_util/sleep.h"
+
+#include <cstdlib>
+
+static int x2 = 0;
+
+int preCalculate(int z) {
+  if (x2 < 10) return 0;
+  return 1;
+}
```
Comment (jonas-kaufmann) [inline src/PerfDetectiveExample/PDEmain.cpp:10]: Might be optimized out, which can be prevented by declaring x2 volatile. Do you plan to also check here how instrumentation behaves under optimization?

**Review by jonas-kaufmann — CHANGES_REQUESTED** [review]
What is the general idea behind this? Do you plan to stress the analysis/instrumentation to check if there are still issues, especially regarding for-loops, which were a problem in the past?

File: src/PerfDetectiveExample/PDEmain.cpp (hunk 2 above)
Comment (vulder) [reply src/PerfDetectiveExample/PDEmain.cpp:10]: No, that's not relevant here. This file just encodes an example from a related paper.
I want to make sure that can handle cases like the ones they explained.

**Review by jonas-kaufmann — APPROVED** [review]

**Pull request merged**